src/adapters/poi_popup.js

```diff
@@ -118,6 +126,7 @@ PoiPopup.prototype.getPopupAnchor = function (event) {
 };
 
 PoiPopup.prototype.close = function () {
+  fire('stop_close_popup_timeout');
   if (this.popupHandle) {
     this.popupHandle.remove();
     this.popupHandle = null;
```

Comment:
I think there is a subtle race condition here, when "mouseleave" occurs during the request to the server (after `timeOutHandler` is executed and before the popup is actually open).

`showPopup()` will call `close()` and cancel the `closeTimeoutHandler` here, whereas the mouse has already left the layer. So, the popup is never closed.

Maybe it would be more appropriate to check that `this.popupHandle` exists for this call too:
```suggestion
PoiPopup.prototype.close = function () {
  if (this.popupHandle) {
    fire('stop_close_popup_timeout');
    this.popupHandle.remove();
    this.popupHandle = null;
```